### PR TITLE
Refactors environment cloner to directly work with physx and usd apis, removing isaac-sim cloner dependency

### PR DESCRIPTION
**Pull request by ooctipus**
# Description
This PR refactors cloner logic into usd_replicate, physx_replicate, (also test to work for newton_replicate but not include in this PR)

At the highest level, This PR decouples cloning responsibility away from spawner, and delegate it to a dedicated lab cloner. The only job for spawner is to spawn prototype, and if any cloning is intended, one needs to use the cloner(either USD cloner, Physx Cloner, or Newton Cloner, whether that be homogeneous cloning or heterogeneous cloning).

In detail this PR
- Adds heterogeneous cloning to InteractiveScene with per-object replication when envs differ, improving PhysX cloning time for heterogeneous scenes while perserving whole env replication in homogenous setting.
- Introduces scene-level flag InteractiveSceneCfg.random_heterogenous_cloning (defaults True) to control prototype assignment:
---True: randomly assigns a prototype per env.
---False: uses round‑robin (env_index % num_prototypes) for determinism (previous behavior).

- Removes hard dependency on the Isaac Sim Cloner; uses internal replicators usd_replicate and physx_replicate.
Deprecates random_choice in MultiAssetSpawnerCfg and MultiUsdFileCfg in favor of the scene-level flag for consistent, scalable control.

While this PR tries to preserve as much original behavior as possible,
with new refactors, it makes cloning with explicit constraints and composition, even proportion very easy, and I will be working on that as next goal.

# Perf

Runing `scripts/demos/multi_asset.py` with 8192 envs

Before:
>[INFO] Time to create scene:  : 29.790888 seconds
>[INFO] Time to randomize scene:  : 0.161381 seconds
>[INFO] to start Simulation:  : 27.550733 seconds

After:
>[INFO] Time to create scene:  : 31.276912 seconds
>[INFO] Time to randomize scene:  : 0.004585 seconds
>[INFO] Time to start Simulation:  : 5.232237 seconds  <--------------- much quicker

## Type of change

- New feature (non-breaking change which adds functionality)
- Breaking change (existing functionality will not work without user modification)

## Screenshots

Please attach before and after screenshots of the change if applicable.

## Checklist

- [x] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab/isaaclab/cloner/cloner_utils.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+
+        # Spawn the first instance of clones from prototypes, then deactivate the prototypes, those first instances
+        # will be served as sources for usd and physx replication.
+        proto_idx = clone_masking.to(torch.int32).argmax(dim=1)
+        proto_mask = torch.zeros_like(clone_masking)
+        proto_mask.scatter_(1, proto_idx.view(-1, 1).to(torch.long), clone_masking.any(dim=1, keepdim=True))
+        usd_replicate(stage, src_paths, dest_paths, world_indices, proto_mask)
+        stage.GetPrimAtPath(cfg.template_root).SetActive(False)
+        replicate_args = [clone_path_fmt.format(0)], [clone_path_fmt], world_indices, clone_masking[0].unsqueeze(0)
+        get_pos = lambda path: stage.GetPrimAtPath(path).GetAttribute("xformOp:translate").Get()  # noqa: E731
```
Comment (kellyguo11) [inline source/isaaclab/isaaclab/cloner/cloner_utils.py:65]: is it ever possible that the prim doesn't have a translate attribute?

File: source/isaaclab/isaaclab/cloner/cloner_utils.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+            dest.append(prototype_root.replace(cfg.template_root, clone_path_fmt))
+
+        src_paths, dest_paths, clone_masking = make_clone_plan(src, dest, num_clones, cfg.clone_strategy, cfg.device)
+
+        # Spawn the first instance of clones from prototypes, then deactivate the prototypes, those first instances
+        # will be served as sources for usd and physx replication.
+        proto_idx = clone_masking.to(torch.int32).argmax(dim=1)
+        proto_mask = torch.zeros_like(clone_masking)
+        proto_mask.scatter_(1, proto_idx.view(-1, 1).to(torch.long), clone_masking.any(dim=1, keepdim=True))
+        usd_replicate(stage, src_paths, dest_paths, world_indices, proto_mask)
```
Comment (kellyguo11) [inline source/isaaclab/isaaclab/cloner/cloner_utils.py:63]: does this mean we always keep the prototype instance available in USD? but they might not be cloned in USD later on?

File: source/isaaclab/isaaclab/cloner/cloner_utils.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        physicsscene_path: Path to PhysicsScene prim.
+        collision_root_path: Root scope for collision groups.
+        prim_paths: Per-clone prim paths.
+        global_paths: Optional global-collider paths.
+
+    Returns:
+        None
+    """
+
+    physx_scene = PhysxSchema.PhysxSceneAPI(stage.GetPrimAtPath(physicsscene_path))
```
Comment (kellyguo11) [inline source/isaaclab/isaaclab/cloner/cloner_utils.py]: would this PhysxSchema reference cause issues in kitless path? do we need to do this for Newton as well or does Newton already handle this with the worlds?

File: source/isaaclab/isaaclab/sim/spawners/wrappers/wrappers.py [hunk 4]
```diff
@@ -44,32 +46,19 @@ def spawn_multi_asset(
...
+    split_path = prim_path.split("/")
+    prefix_path, base_name = "/".join(split_path[:-1]), split_path[-1]
+    if ".*" not in base_name:
+        raise ValueError(
+            f" The base name '{base_name}' in the prim path '{prim_path}' must contain '.*' to indicate"
+            " the path each individual multiple-asset to be spawned."
+        )
+    if cfg.random_choice:
+        logger.warning(
+            "`random_choice` parameter in `spawn_multi_asset` is deprecated, and nothing will happen. "
```
Comment (kellyguo11) [inline source/isaaclab/isaaclab/sim/spawners/wrappers/wrappers.py:58]: hmm maybe that's not the purpose of deprecation. I think we should still preserve behavior when we deprecate things, but it'll just throw a warning to let people know what the new way should be.

File: source/isaaclab/isaaclab/sim/utils/prims.py [hunk 5]
```diff
@@ -727,35 +726,31 @@ def wrapper(prim_path: str | Sdf.Path, cfg: SpawnerCfg, *args, **kwargs):
...
-                    replicate_physics=replicate_physics,
-                    copy_from_source=cfg.copy_from_source,
-                    clone_in_fabric=clone_in_fabric,
-                )
+        if len(source_prim_paths) > 1:
+            # lazy import to avoid circular import
+            from isaaclab.cloner import usd_replicate
+
+            formattable_path = f"{root_path.replace('.*', '{}')}/{asset_path}"
+            usd_replicate(
```
Comment (kellyguo11) [inline source/isaaclab/isaaclab/sim/utils/prims.py:705]: why is this always usd_replicate? is this path still triggered on spawning of every asset?

File: source/isaaclab/isaaclab/utils/assets.py [hunk 6]
```diff
@@ -39,6 +39,9 @@
 ISAACLAB_NUCLEUS_DIR = f"{ISAAC_NUCLEUS_DIR}/IsaacLab"
 """Path to the ``Isaac/IsaacLab`` directory on the NVIDIA Nucleus Server."""
 
+ISAACLAB_LOCAL_DIR = os.path.normpath(os.path.join(os.path.dirname(__file__), "../../../isaaclab_assets/data"))
```
Comment (kellyguo11) [inline source/isaaclab/isaaclab/utils/assets.py]: what is this used for?

File: source/isaaclab/test/performance/test_robot_load_performance.py [hunk 7]
```diff
@@ -39,15 +43,32 @@ def test_robot_load_performance(test_config, device):
...
-            source_prim_path=target_paths[0],
-            prim_paths=target_paths,
-            replicate_physics=False,
-            copy_from_source=True,
+        stage = sim_utils.get_current_stage()
+
+        # Generate grid positions for environments
+        positions, _ = cloner.grid_transforms(NUM_ENVS, SPACING, device=device)
+
+        # Create environment prims using USD replicate
```
Comment (kellyguo11) [inline source/isaaclab/test/performance/test_robot_load_performance.py:51]: should all the direct environments and standalone scripts also follow this cloning workflow?

Comment (ooctipus) [reply source/isaaclab/test/performance/test_robot_load_performance.py:51]: they don't need any changes if they are using interactive scene. This style only needed for those not using interactive scene.

File: source/isaaclab/isaaclab/utils/assets.py (hunk 6 above)
Comment (ooctipus) [reply source/isaaclab/isaaclab/utils/assets.py]: sorry let me remove this

File: source/isaaclab/isaaclab/cloner/cloner_utils.py (hunk 2 above)
Comment (ooctipus) [reply source/isaaclab/isaaclab/cloner/cloner_utils.py:63]: no we delete/deactivate protype instance once cloned to first instance, and we always do that step

File: source/isaaclab/isaaclab/sim/spawners/wrappers/wrappers.py (hunk 4 above)
Comment (AntoineRichard) [reply source/isaaclab/isaaclab/sim/spawners/wrappers/wrappers.py:58]: Agreed. I'll create a ticket so we remember this. And we can merge that PR in

File: source/isaaclab/test/sim/test_cloner.py
Comment (AntoineRichard) [inline source/isaaclab/test/sim/test_cloner.py:1]: I feel like the PhysX replicate should have it's own tests in PhysX

File: source/isaaclab/test/performance/test_robot_load_performance.py (hunk 7 above)
Comment (AntoineRichard) [reply source/isaaclab/test/performance/test_robot_load_performance.py:51]: Maybe we could provide an easy utility do to the cloning without the user having to manually do all of this?

File: source/isaaclab/isaaclab/sim/utils/prims.py
Comment (AntoineRichard) [inline source/isaaclab/isaaclab/sim/utils/prims.py:624]: Do we want to move this function outside of this utility and put it with the cloning stuff?

File: source/isaaclab/isaaclab/cloner/cloner_utils.py (hunk 1 above)
Comment (AntoineRichard) [reply source/isaaclab/isaaclab/cloner/cloner_utils.py:65]: We have a utility function to normalize the xform stack. Maybe we could apply it by default to ensure the XformOps are what we expect?

File: source/isaaclab/isaaclab/sim/spawners/wrappers/wrappers.py (hunk 4 above)
Comment (AntoineRichard) [reply source/isaaclab/isaaclab/sim/spawners/wrappers/wrappers.py:58]: https://github.com/isaac-sim/IsaacLab/issues/4693

File: source/isaaclab/test/sim/test_cloner.py
Comment (AntoineRichard) [reply source/isaaclab/test/sim/test_cloner.py:1]: https://github.com/isaac-sim/IsaacLab-Internal/issues/801

File: source/isaaclab/test/performance/test_robot_load_performance.py (hunk 7 above)
Comment (AntoineRichard) [reply source/isaaclab/test/performance/test_robot_load_performance.py:51]: https://github.com/isaac-sim/IsaacLab/issues/4695

File: source/isaaclab/test/sim/test_cloner.py
Comment (ooctipus) [reply source/isaaclab/test/sim/test_cloner.py:1]: yep I can do that

**Comment by ooctipus** [comment]
```
| /source/isaaclab_tasks/test/benchmarking/test_environments_training.py                | FAILED |  2627.50 |  6/516  |
| /source/isaaclab_mimic/test/test_curobo_planner_franka.py                             | FAILED |     0.00 |   0/1   |
| /source/isaaclab_mimic/test/test_curobo_planner_cube_stack.py                         | FAILED |     0.00 |   0/1   |
| /source/isaaclab_mimic/test/test_generate_dataset_skillgen.py                         | FAILED |    18.48 |   0/1   |
| /source/isaaclab/test/performance/test_kit_startup_performance.py                     | FAILED |    16.51 |   0/1   |
| /source/isaaclab/test/sensors/test_multi_tiled_camera.py                              | FAILED |     7.62 |   6/11  |
| /source/isaaclab/test/sensors/test_multi_mesh_ray_caster_camera.py                    | FAILED |    20.14 |  16/20  |
| /source/isaaclab/test/sensors/test_tiled_camera.py                                    | FAILED |    52.44 |  57/61  |
| /source/isaaclab/test/sensors/test_ray_caster_camera.py                               | FAILED |    22.63 |  18/21  |
| /source/isaaclab/test/controllers/test_pink_ik.py                                     | FAILED |   199.81 |  17/24  |
| /source/isaaclab/test/controllers/test_operational_space.py                           | FAILED |   131.33 |  16/17  |
```
